docs/create_your_own_extension_layer.md

```diff
@@ -0,0 +1,177 @@
+# How to build new layer
+
+To create your own extension layer, compile the extension and any required libraries on Bref environment with Docker.
+It picks up pgsql extension as an example, so please replace the name appropriately.
+
+First create new extension directory in `layers/` and move there.
+
+```bash
+$ mkdir layers/pgsql && cd $_
+```
+
+Create `Dockerfile` such as follows.
+
+```Dockerfile
+ARG PHP_VERSION
+FROM bref/build-php-$PHP_VERSION AS ext
+
+#
+# Add commands to build PHP extensions here.
+#
+
+# Build the final image from the lambci image that is close to the production environment
+FROM lambci/lambda:provided
+
+#
+# Add commands to copy files that requried for final image.
+#
+```
+
+Then you need to add two part of Dockerfile, the following snippets are examples of build and file copy parts respectively.
+
+```Dockerfile
```

Comment:
Maybe introduce this part of the dockerfile a little. Say that we need to install the libraries and fix things to eventually install the php extension